### PR TITLE
Adding http version and streaming info to logging output

### PR DESCRIPTION
**Pull request by samsp-msft**
Final part of #600

### CODE REVIEW

File: src/ReverseProxy/Forwarder/HttpForwarder.cs [hunk 1]
```diff
@@ -771,12 +771,17 @@ public static void HttpDowngradeDetected(ILogger logger)
...
+            public static void Proxying(ILogger logger, HttpRequestMessage msg, bool isStreamingRequest)
             {
                 // Avoid computing the AbsoluteUri unless logging is enabled
                 if (logger.IsEnabled(LogLevel.Information))
                 {
-                    _proxying(logger, targetUrl.AbsoluteUri, null);
+#if NET
+                    var versionInfo = $"http{msg.Version} {msg.VersionPolicy}";
+#else
+                    var versionInfo = $"http {msg.Version}";
```
Comment (Tratcher) [inline src/ReverseProxy/Forwarder/HttpForwarder.cs:782]:
```suggestion
                    var versionInfo = $"http/{msg.Version} {msg.VersionPolicy}";
#else
                    var versionInfo = $"http/{msg.Version}";
```

**Review by Tratcher — APPROVED** [review]
Can you show what the new log output looks like now? I don't think Version.ToString formats the way you want by default.
You can borrow GetHttpProtocol:
https://github.com/dotnet/aspnetcore/blob/393a86c0e7218964e557adeb833010e1482bb96b/src/Http/Http.Abstractions/src/HttpProtocol.cs#L112
And put it here:
https://github.com/microsoft/reverse-proxy/blob/2015d992e0061c6849fbe11c62311161b3399210/src/ReverseProxy/Forwarder/ProtocolHelper.cs#L9

**Comment by samsp-msft** [comment]
> Can you show what the new log output looks like now? I don't think Version.ToString formats the way you want by default. You can borrow GetHttpProtocol: https://github.com/dotnet/aspnetcore/blob/393a86c0e7218964e557adeb833010e1482bb96b/src/Http/Http.Abstractions/src/HttpProtocol.cs#L112 And put it here:
> 
> https://github.com/microsoft/reverse-proxy/blob/2015d992e0061c6849fbe11c62311161b3399210/src/ReverseProxy/Forwarder/ProtocolHelper.cs#L9

Current output:

```console
info: Yarp.ReverseProxy.Forwarder.HttpForwarder[9]
      Proxying to http://www.example.com/ - http2.0 RequestVersionOrLower no-streaming
```

How exactly do we want it to read? Options:

*      Proxying to http://www.example.com/ - HTTP/2 RequestVersionOrLower no-streaming
*      Proxying to http://www.example.com/ - http2 ≤  false
*      Proxying to http://www.example.com/ - http≤2  no-stream

**Comment by Tratcher** [comment]
> ` Proxying to http://www.example.com/ - HTTP/2 RequestVersionOrLower no-streaming`

Looks good.

File: src/ReverseProxy/Forwarder/HttpForwarder.cs [hunk 2]
```diff
@@ -756,10 +756,10 @@ private static class Log
                 EventIds.HttpDowngradeDetected,
                 "The request was downgraded from HTTP/2.");
 
-            private static readonly Action<ILogger, string, Exception?> _proxying = LoggerMessage.Define<string>(
+            private static readonly Action<ILogger, string, string, string, Exception?> _proxying = LoggerMessage.Define<string, string, string>(
                 LogLevel.Information,
                 EventIds.Forwarding,
-                "Proxying to {targetUrl}");
+                "Proxying to {targetUrl} - {verson} {isStreaming}");
```
Comment (Kahbazi) [inline src/ReverseProxy/Forwarder/HttpForwarder.cs:762]:
```suggestion
                "Proxying to {targetUrl} - {version} {isStreaming}");
```

Comment (Kahbazi) [reply src/ReverseProxy/Forwarder/HttpForwarder.cs:762]: Shouldn't `isStreaming` be `bool` so it's easier to find it when using structured logs?

File: src/ReverseProxy/Forwarder/HttpForwarder.cs [hunk 3]
```diff
@@ -771,12 +771,17 @@ public static void HttpDowngradeDetected(ILogger logger)
...
 
-            public static void Proxying(ILogger logger, Uri targetUrl)
+            public static void Proxying(ILogger logger, HttpRequestMessage msg, bool isStreamingRequest)
             {
                 // Avoid computing the AbsoluteUri unless logging is enabled
                 if (logger.IsEnabled(LogLevel.Information))
                 {
-                    _proxying(logger, targetUrl.AbsoluteUri, null);
+#if NET
+                    var versionInfo = $"http/{msg.Version} {msg.VersionPolicy}";
```
Comment (Kahbazi) [inline src/ReverseProxy/Forwarder/HttpForwarder.cs:780]: Why not create another property for `VersionPolicy` in logs? When `Version` and `VersionPolicy` are separate properties it's easier to have different queries based on `Version` alone.

**Review by Tratcher — COMMENTED** [review]
I pushed some minor changes to this branch. The output now looks like this:
`Proxying to https://example.com/ HTTP/2 RequestVersionOrLower no-streaming`

File: src/ReverseProxy/Forwarder/HttpForwarder.cs (hunk 2 above)
Comment (Kahbazi) [reply src/ReverseProxy/Forwarder/HttpForwarder.cs]: In the log message "streaming" and "non-streaming" is a better choice but I thought in a structured log view like Elasticsearch, storing bool needs less space than string also writing a query like `isStreaming = true` is better than `isStreaming = streaming`.

Comment (Tratcher) [reply src/ReverseProxy/Forwarder/HttpForwarder.cs]: Can you show an example of that?

**Pull request merged**